ql/src/test/queries/clientpositive/multi_insert_gby5.q

```diff
@@ -0,0 +1,16 @@
+set hive.cbo.fallback.strategy=NEVER;
+
+CREATE TABLE tbl1 (key int, f1 int);
+CREATE TABLE tbl2 (f1 int) PARTITIONED BY (key int);
+
+EXPLAIN FROM (SELECT key, f1 FROM tbl1 WHERE key=5) a
```

Comment:
The plan looks valid but just to be on the safe side let's add some data in the table to ensure we are getting back correct results since this is something that never worked before.

Reply:
Ok, I added some data.  The query itself is kinda weird, as you mentioned in the comments.

I put one row with "key = 5" which should be selected twice.  One row in the new table will contain key=6 because it is explicitly put into the partition.  The value is the same on both rows.

Definitely a weird query, but I think it shows the right data is being filtered and inserted.